main.tf

```diff
@@ -103,7 +103,7 @@ resource "azuread_application" "client" {
 
     resource_access {
       # Server app Oauth2 permissions id
-      id   = lookup(azuread_application.server.oauth2_permissions[0], "id")
+      id   = [for permission in azuread_application.server.oauth2_permissions : permission.id][0]
```

Comment:
The current code seens to be working just fine. It seems like the follwing code would be "cleaner":
``` id = tolist(azuread_application.server.oauth2_permissions)[0].id```

Reply:
The code provided was taken from the release notes at https://github.com/terraform-providers/terraform-provider-azuread/blob/master/CHANGELOG.md#0100-june-05-2020